File: src/main/java/com/esprit/models/Event.java
package com.esprit.models;

import java.util.Objects;

public class Event {
    private int id;
    private String date;
    private float prixdupass;
    private String lieu;
    private String horaire;
    private String image;
    private double revenu;
    public Event(int id, String date, float prixdupass, String lieu, String horaire ,String image) {
        this.id = id;
        this.date = date;
        this.prixdupass = prixdupass;
        this.lieu = lieu;
        this.horaire = horaire;
        this.image=image;
    }

    public Event(String date, float prixdupass, String lieu, String horaire,String image) {
        this.date = date;
        this.prixdupass = prixdupass;
        this.lieu = lieu;
        this.horaire = horaire;
        this.image=image;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public int getId() {
        return id;
    }

    public String getDate() {
        return date;
    }

    public float getPrixdupass() {
        return prixdupass;
    }

    public String getLieu() {
        return lieu;
    }

    public String getHoraire() {
        return horaire;
    }

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    public void setPrixdupass(float prixdupass) {
        this.prixdupass = prixdupass;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setHoraire(String horaire) {
        this.horaire = horaire;
    }
    public double getRevenu() {
        return revenu;
    }

    public void setRevenu(double revenu) {
        this.revenu = revenu;
    }
    @Override
    public String toString() {
        return "Event{" +
                "id=" + id +
                ", date='" + date + '\'' +
                ", prixdupass=" + prixdupass +
                ", lieu='" + lieu + '\'' +
                ", horaire='" + horaire + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Event event = (Event) o;
        return id == event.id && Float.compare(prixdupass, event.prixdupass) == 0 && Objects.equals(date, event.date) && Objects.equals(lieu, event.lieu) && Objects.equals(horaire, event.horaire);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, date, prixdupass, lieu, horaire);
    }


}

File: src/Entity/Event.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\EventRepository;

#[ORM\Entity(repositoryClass: EventRepository::class)]
#[ORM\Table(name: 'event')]
class Event
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): self
    {
        $this->id = $id;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: false)]
    private ?string $date = null;

    public function getDate(): ?string
    {
        return $this->date;
    }

    public function setDate(string $date): self
    {
        $this->date = $date;
        return $this;
    }

    #[ORM\Column(type: 'float', nullable: false)]
    private ?float $prixdupass = null;

    public function getPrixdupass(): ?float
    {
        return $this->prixdupass;
    }

    public function setPrixdupass(float $prixdupass): self
    {
        $this->prixdupass = $prixdupass;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: true)]
    private ?string $lieu = null;

    public function getLieu(): ?string
    {
        return $this->lieu;
    }

    public function setLieu(?string $lieu): self
    {
        $this->lieu = $lieu;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: false)]
    private ?string $horaire = null;

    public function getHoraire(): ?string
    {
        return $this->horaire;
    }

    public function setHoraire(string $horaire): self
    {
        $this->horaire = $horaire;
        return $this;
    }

    #[ORM\Column(type: 'string', nullable: false,length:255)]
    private ?string $image = null;

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;
        return $this;
    }

    #[ORM\OneToMany(targetEntity: Activiteevent::class, mappedBy: 'event')]
    private Collection $activiteevents;

    /**
     * @return Collection<int, Activiteevent>
     */
    public function getActiviteevents(): Collection
    {
        if (!$this->activiteevents instanceof Collection) {
            $this->activiteevents = new ArrayCollection();
        }
        return $this->activiteevents;
    }

    public function addActiviteevent(Activiteevent $activiteevent): self
    {
        if (!$this->getActiviteevents()->contains($activiteevent)) {
            $this->getActiviteevents()->add($activiteevent);
        }
        return $this;
    }

    public function removeActiviteevent(Activiteevent $activiteevent): self
    {
        $this->getActiviteevents()->removeElement($activiteevent);
        return $this;
    }

    #[ORM\OneToMany(targetEntity: Client::class, mappedBy: 'event')]
    private Collection $clients;

    public function __construct()
    {
        $this->activiteevents = new ArrayCollection();
        $this->clients = new ArrayCollection();
    }

    /**
     * @return Collection<int, Client>
     */
    public function getClients(): Collection
    {
        if (!$this->clients instanceof Collection) {
            $this->clients = new ArrayCollection();
        }
        return $this->clients;
    }

    public function addClient(Client $client): self
    {
        if (!$this->getClients()->contains($client)) {
            $this->getClients()->add($client);
        }
        return $this;
    }

    public function removeClient(Client $client): self
    {
        $this->getClients()->removeElement($client);
        return $this;
    }
    public function __toString(): string
    {
        try {
            // Vérification plus robuste
            $dateStr = (method_exists($this, 'getDate') && $this->getDate() instanceof \DateTimeInterface) 
                ? $this->getDate()
                : null;
    
            return $dateStr ? $dateStr.' - Événement #'.$this->getId() 
                          : 'Événement #'.$this->getId();
            
        } catch (\Exception $e) {
            // Fallback sécurisé en cas d'erreur
            return 'Événement #'.$this->getId();
        }
    }
}
